### PR TITLE
refactor(components): [pagination] use JSX in Unit test

### PR DESCRIPTION
**Pull request by Chuck-Lau**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by ryuhangyeong** [comment]
Can you please resolve the conflict?

**Comment by Tsong-LC** [comment]
> Can you please resolve the conflict?

OK！Thanks for your review.

File: packages/components/pagination/__tests__/pagination.test.tsx [hunk 1]
```diff
@@ -53,16 +48,10 @@ describe('Pagination', () => {
...
-            return h(Pagination, {
-              total: 100,
-              currentPage: 1,
-            })
-          }
-        },
-      })
+      const wrapper = mount(
+        <Pagination total={100} currentPage={1}></Pagination>
+      )
```
Comment (ryuhangyeong) [inline packages/components/pagination/__tests__/pagination.test.tsx:53]: Could you change the syntax to `mount(() => )` for consistency with other test statements?

File: packages/components/pagination/__tests__/pagination.test.tsx [hunk 2]
```diff
@@ -83,33 +72,21 @@ describe('Pagination', () => {
...
-            return h(Pagination, {
-              layout: 'sizes, pager',
-              pageSize: 10,
-            })
-          }
-        },
-      })
+      const wrapper = mount(
+        <Pagination layout="sizes, pager" pageSize={10}></Pagination>
+      )
```
Comment (ryuhangyeong) [inline packages/components/pagination/__tests__/pagination.test.tsx:77]: same here

File: packages/components/pagination/__tests__/pagination.test.tsx [hunk 3]
```diff
@@ -143,15 +120,8 @@ describe('Pagination', () => {
...
-      const wrapper = mount({
-        setup() {
-          return () => {
-            return h(Pagination, {
-              total: 100,
-            })
-          }
-        },
-      })
+      const wrapper = mount(<Pagination total={100}></Pagination>)
```
Comment (ryuhangyeong) [inline packages/components/pagination/__tests__/pagination.test.tsx:123]: same here

File: packages/components/pagination/__tests__/pagination.test.tsx [hunk 4]
```diff
@@ -166,51 +136,29 @@ describe('Pagination', () => {
...
-        components: {
-          'el-pagination': Pagination,
-        },
-      }
-      const wrapper = mount(TestComponent)
+      const wrapper = mount(
+        <Pagination layout="slot, prev, pager, next" pageSize={25} total={100}>
+          <span class="slot-test">slot test</span>
+        </Pagination>
+      )
```
Comment (ryuhangyeong) [inline packages/components/pagination/__tests__/pagination.test.tsx:143]: same here

File: packages/components/pagination/__tests__/pagination.test.tsx [hunk 5]
```diff
@@ -344,15 +270,7 @@ describe('Pagination', () => {
...
-      const wrapper = mount({
-        setup() {
-          return () => {
-            return h(Pagination, {
-              total: 100,
-            })
-          }
-        },
-      })
+      const wrapper = mount(<Pagination total={100} />)
```
Comment (ryuhangyeong) [inline packages/components/pagination/__tests__/pagination.test.tsx:273]: same here

File: packages/components/pagination/__tests__/pagination.test.tsx [hunk 6]
```diff
@@ -387,15 +305,7 @@ describe('Pagination', () => {
...
```
Comment (ryuhangyeong) [inline packages/components/pagination/__tests__/pagination.test.tsx:308]: same here

File: packages/components/pagination/__tests__/pagination.test.tsx [hunk 7]
```diff
@@ -413,16 +323,7 @@ describe('Pagination', () => {
...
-        setup() {
-          return () => {
-            return h(Pagination, {
-              total: 100,
-              disabled: true,
-            })
-          }
-        },
-      })
+      const wrapper = mount(<Pagination total={100} disabled={true} />)
```
Comment (ryuhangyeong) [inline packages/components/pagination/__tests__/pagination.test.tsx:326]: same here

**Review by ryuhangyeong — APPROVED** [review]

**Comment by ryuhangyeong** [comment]
Thank you!

**Pull request merged**